### ISSUE TITLE
Apple warning `UIWebView`

### ISSUE THREAD

**Issue by cipriancaba** [invertase/react-native-firebase#2535]
## Issue

Apple started issuing warnings for libs that use the deprecated `UIWebView`. Firebase Authentication depends on `FirAuth` that references this old api

Is is to be expected for Apple to start rejecting apps that use this api.

A commit has already been pushed to firebase auth repo
https://github.com/firebase/firebase-ios-sdk/pull/3723

React native discussion
https://github.com/react-native-community/react-native-webview/issues/819

We should aim to push an updated dependency as soon as Firebase Auth releases a new version

---

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
pod 'RNFBAuth', :path => "#{nm_path}@react-native-firebase/auth/ios"
```

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
react-native info
warn The following packages use deprecated "rnpm" config that will stop working from next release:
  - react-native-maps: https://npmjs.com/package/react-native-maps
Please notify their maintainers about it. You can find more details at https://github.com/react-native-community/cli/blob/master/docs/configuration.md#migration-guide.
info Fetching system and libraries information...
System:
    OS: macOS 10.14.6
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
    Memory: 8.65 GB / 32.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.6.0 - /usr/local/bin/node
    Yarn: 1.16.0 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 12.4, macOS 10.14, tvOS 12.4, watchOS 5.3
    Android SDK:
      API Levels: 23, 24, 25, 26, 27, 28
      Build Tools: 23.0.1, 24.0.3, 25.0.2, 25.0.3, 26.0.1, 26.0.2, 26.0.3, 27.0.3, 28.0.1, 28.0.3
      System Images: android-24 | Google APIs Intel x86 Atom, android-27 | Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom
  IDEs:
    Android Studio: 3.4 AI-183.6156.11.34.5522156
    Xcode: 10.3/10G8 - /usr/bin/xcodebuild
  npmPackages:
    react-native: 0.60.4 => 0.60.4
```

- **Platform that you're experiencing the issue on**:
  - [x] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `e.g. 5.4.3`
- **`Firebase` module(s) you're using that has the issue:**
  - `e.g. Instance ID`
- **Are you using `TypeScript`?**
  - `Y/N`
  
</p>
</details>

---

Think `react-native-firebase` is great? Please consider supporting all of the project maintainers and contributors by donating via our [Open Collective](https://opencollective.com/react-native-firebase/donate) where all contributors can submit expenses. [[Learn More]](https://invertase.io/oss/react-native-firebase/contributing/donations-expenses)

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by Ehesp** [comment]
Thanks for the report. Once the SDK version is released we'll get it tested and updated with a release.

**Comment by mikehardy** [comment]
I'm not actually sure there is anything to do? Once it's released you just `pod outdated` to see it and `pod update` to install it? v6 and v5.x.x of react-native-firebase support the current release line of the underlying SDKs

**Comment by mikehardy** [comment]
Just watch here since they cherry-picked to their 6.8.0 pending release. Once you see 6.8.0 install it https://firebase.google.com/support/release-notes/ios

Also, this is an App Store warning, not a fail. They'll keep accepting these submissions for now. It's also affecting react-native-webview and react-native-device-info at minimum.

**Comment by cipriancaba** [comment]
I assume that the auth package needs to update this dependency https://github.com/invertase/react-native-firebase/blob/ec85e5e3f58d603b490d00de4f575da13ae2d7b2/packages/auth/RNFBAuth.podspec#L20

Re react-native-webview and react-native-device-info they both posted fixes

**Comment by Ehesp** [comment]
Yeah on v6 we hard set the versions so that will need changing, v5 should be do-able via updating pods yourself.

**Comment by mikehardy** [comment]
oh my - I haven't used v6 yet, the dependencies are coded into each subspec in the packages? they are not configurable in the main app podspec :scream: 

I am sure there are reasons for that design (like, making it so people don't have to think about adding dependencies...) but I'd hack up a pre script in my main app podspec to control the dependencies personally. sorta like this but moving versions or similar https://guides.cocoapods.org/syntax/podfile.html#pre_install - but I'm pretty touchy about being able to control my own versions

**Comment by mikehardy** [comment]
also I can't imagine trying to track the ios sdk releases? e.g., firestore 6.7.1 is out and fixes a crash bug, if they release that sdk over the weekend is there some expectation you cut a release to match contemporaneously? or recommend patch-package? :thinking: - seems like a really tight coupling on something otherwise not in control

**Comment by Ehesp** [comment]
Let me get back to you on the specifics. You can override the RNFB versions yourself though. I'll speak with @Salakar - probably one to document too.

**Comment by quachsimon** [comment]
What is the current status of this warning? Our team is also running into this and points to FIRAuth. We are on v5.4.3 of RNFirebase. 

Would we simply update the Podfile `Firebase/Auth` dependency?

**Comment by mikehardy** [comment]
They are fixing things on a very rapid basis - for instance a difference between Firebase iOS SDK release 6.8.0 and 6.8.1 was related to this even, https://firebase.google.com/support/release-notes/ios#version_681_-_september_10_2019 - just a few days ago.

My best recommendation is to get completely current, even though I recognize this means ingesting some breaking changes - read the release notes carefully for react-native-firebase v5.5.0 (on your way to v5.5.6 and Pods 6.8.1)

**Comment by mikehardy** [comment]
@quachsimon that was for you - I believe the original reporter was for react-native-firebase v6, which just had a "pod overrides" PR proposed by @Ehesp #2589

**Comment by quachsimon** [comment]
Thanks @mikehardy. Will look into updating to react-native-firebase v5.5.6. 

After that, I can just change my `Firebase/Auth` pod to `6.8.1`? From the [docs](https://rnfirebase.io/docs/v5.x.x/releases/v5.5.x#iOS---Update-Firebase-SDKs) it suggests to lock in `6.2.0` for v5.5.x. This should be ok?

**Comment by mikehardy** [comment]
Oh dang - my documentation is in conflict. My goal with v5.x react-native-firebase maintenance is to allow use of the most up to date underlying SDKs, even if that means small breaking changes in RNFB. I'm personally on 6.8.1 and it's okay.

I updated this table to just be '6.x' for the pods https://rnfirebase.io/docs/v5.x.x/getting-started#Supported-versions---React-Native-/-Firebase and I will update that doc to say that it is best to pin your version yes, but also periodically run `pod outdated` and update the pods assuming they haven't done breaking changes (so, in the 6.x ios pod range)

So TLDR - 6.8.1 should be okay. If you are on react-native 0.60+ you must use the *branch* v5.x.x from git as there is a patch needed to make it work with AndroidX that we haven't released yet https://github.com/invertase/react-native-firebase/pull/2476

**Comment by mikehardy** [comment]
See if this is more clear: https://github.com/invertase/react-native-firebase-docs/pull/217 - if you like it, you can say so on the PR and I'll merge. Open to any wording suggestions of course

**Comment by quachsimon** [comment]
Amazing @mikehardy! Will update in the new version and update afterwards!

**Comment by LukePenkava** [comment]
@mikehardy Hi, i am on RN 0.59.10 using react-native-firebase ~5.5.5. I am getting the UIWebView warning from Apple as well. I realy cant transfer to RN 0.60 for production reasons. I am just worried if i will update to latest React Native Firebase i will be forced to switch to 0.60. Let me please know if there is some fix for this Apple warning and staying on RN 0.59.10. Thank you.

**Comment by mikehardy** [comment]
no worries, I test it with rn59 and I believe it is still working - demonstration here: https://github.com/mikehardy/rnfbdemo/blob/master/make-demo-rn59.sh

You will still have other work to do as well - I maintain react-native-device-info and it isn't fixed with regard to UIWebView until v3.x.x series, react-native-webview also only just recently. I believe both of those made the transition to AndroidX as well but...I also maintain jetifier, and using `jetifier -r` (for reverse mode) you should be able to stay on RN0.59 for a bit longer. You'll have to move foward soon though - this is a very tough divide to straddle (RN59 / RN60)

**Comment by LukePenkava** [comment]
@mikehardy Thanks, yeah, i just need to stay on 0.59.10 since i am like 2 weeks from release. So just to clarify, i should update to 5.5.6 and update my pods right ? I am on these 

```
  pod 'Firebase/Core', '~> 6.3.0'
  pod 'Firebase/Auth', '~> 6.3.0'
  pod 'Firebase/Database', '~> 6.3.0'
  pod 'Firebase/Messaging', '~> 6.3.0'
```

I am not sure to what version i should update those to. I think you mentioned 6.8.1 is ok with 5.5.6 and RN 0.59.10 ?

**Comment by quachsimon** [comment]
@LukePenkava yes, we are on 5.5.6 and have updated our pods to `~> 6.8.1` and on 0.59.10 and everything is working great!

**Comment by LukePenkava** [comment]
@quachsimon great, thank you

**Comment by mikehardy** [comment]
yep - I've heard no bad reports on 0.59.10 + pods 6.8.1 (or 6.9.0 now) and 5.5.6

only for people that move from rn 0.59.10 to rn 0.60+ they may need an unreleased react-native-firebase patch that is committed to v5.x.x just not out as a 5.5.7 yet

good luck with the release!

**Comment by LukePenkava** [comment]
@mikehardy thanks :) I can confirm that it works fine with 5.5.6 and 6.8.1. The issue with UIWebView when uploading to Appstore is still there tho. I searched through my RN project and it seems that all references to UIWebView are gone now ( before a lot of references to Firebase ), yet the Apple still gives the warning. It might be some other completely issue unrelated to React-Native-Firebase, but just wondering if you perhaps know what might be causing the issue.

**Comment by mikehardy** [comment]
Hmm - I know of lots of sources of it. For instance I maintain react-native-device-info and that was a big rush fixing it, you need v3.x there. react-native-webview updated. You got the firebase ios sdk updated so that should be settled (unless maybe you need to deep clean your build system with a react-native-clean-project run where you say yes to "wipe iOS build"). If you look at the react-native-webview repo, there is an issue running there where a few people posted good troubleshooting tips to find other sources of UIWebView symbol references

**Comment by LukePenkava** [comment]
@mikehardy I will look into finding other sources of UIWebView. If the update should have solved the UIWebView issue for Firebase, then it must be something else. Thanks again.

**Comment by mikehardy** [comment]
This is resolved I believe

**Comment by stepupithub** [comment]
Installing Firebase (6.14.0)
Installing FirebaseAnalytics (6.1.7)
Installing FirebaseAnalyticsInterop (1.4.0)
Installing FirebaseAuth (6.4.1)
Installing FirebaseAuthInterop (1.0.0)
Installing FirebaseCore (6.5.0)
Installing FirebaseCoreDiagnostics (1.1.2)
Installing FirebaseCoreDiagnosticsInterop (1.1.0)
Installing FirebaseFirestore (1.8.3)
Installing FirebaseInstanceID (4.2.8)
Installing FirebaseMLCommon (0.19.0)
Installing FirebaseMLVision (0.19.0)
Installing FirebaseMLVisionBarcodeModel (0.19.0)
Installing FirebaseMessaging (4.1.10)
Installing FirebaseStorage (3.4.3)


how to change firebaseauth to 6.8.1 ???
package.json 
react-native-firebase "~5.5.6"

**Comment by stepupithub** [comment]
Installing Firebase (6.14.0)
Installing FirebaseAnalytics (6.1.7)
Installing FirebaseAnalyticsInterop (1.4.0)
Installing FirebaseAuth (6.4.1)
Installing FirebaseAuthInterop (1.0.0)
Installing FirebaseCore (6.5.0)
Installing FirebaseCoreDiagnostics (1.1.2)
Installing FirebaseCoreDiagnosticsInterop (1.1.0)
Installing FirebaseFirestore (1.8.3)
Installing FirebaseInstanceID (4.2.8)
Installing FirebaseMLCommon (0.19.0)
Installing FirebaseMLVision (0.19.0)
Installing FirebaseMLVisionBarcodeModel (0.19.0)
Installing FirebaseMessaging (4.1.10)
Installing FirebaseStorage (3.4.3)


how to change it plzzz...help me @mikehardy

**Comment by stepupithub** [comment]
this is my pod file 
react-native version 0.59.10

 platform :ios, '9.0'

target 'avunja' do
  # Comment the next line if you don't want to use dynamic frameworks
  #use_frameworks!
pod 'React', :path => '../node_modules/react-native', :subspecs => [
    'Core',
    'CxxBridge',
    'DevSupport',
    # the following ones are the ones taken from "Libraries" in Xcode:
    'RCTAnimation',
    'RCTActionSheet',
    'RCTBlob',
    'RCTGeolocation',
    'RCTImage',
    'RCTLinkingIOS',
    'RCTNetwork',
    'RCTSettings',
    'RCTText',
    'RCTVibration',
    'RCTWebSocket'
  ]

  # Pods for avunja
pod 'yoga', :path => '../node_modules/react-native/ReactCommon/yoga/Yoga.podspec'
pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'
pod 'RNFirebase', :path => "../node_modules/react-native-firebase/ios/"
pod 'react-native-camera', path: '../node_modules/react-native-camera', subspecs: [
  'BarcodeDetectorMLKit'
]
pod 'Firebase/Auth'
pod 'Firebase/Firestore'
pod 'Firebase/Core'
pod 'Firebase/Messaging'
pod 'Firebase/Storage'

pod 'GoogleTagManager'
pod 'Firebase/MLVision'
pod 'Firebase/MLVisionBarcodeModel'

  target 'avunjaTests' do
    inherit! :search_paths
    # Pods for testing
  end

end

post_install do |installer|

  rnfirebase = installer.pods_project.targets.find { |target| target.name == 'RNFirebase' }
  rnfirebase.build_configurations.each do |config|
    config.build_settings['HEADER_SEARCH_PATHS'] = '$(inherited) ${PODS_ROOT}/Headers/Public/**'
  end
end

**Comment by stepupithub** [comment]
where should i change @mikehardy @LukePenkava

**Comment by stepupithub** [comment]
is that okay @mikehardy 

Installing BoringSSL-GRPC (0.0.3)
Installing Crashlytics (3.14.0)
Installing DoubleConversion (1.1.6)
Installing Fabric (1.10.2)
Installing Firebase (6.8.1)
Installing FirebaseAnalytics (6.1.1)
Installing FirebaseAnalyticsInterop (1.4.0)
Installing FirebaseAuth (6.2.3)
Installing FirebaseAuthInterop (1.0.0)
Installing FirebaseCore (6.2.3)
Installing FirebaseCoreDiagnostics (1.1.2)
Installing FirebaseCoreDiagnosticsInterop (1.1.0)
Installing FirebaseFirestore (1.5.1)
Installing FirebaseInstanceID (4.2.7)
Installing FirebaseMLCommon (0.18.0)
Installing FirebaseMLVision (0.18.0)
Installing FirebaseMLVisionBarcodeModel (0.18.0)

**Comment by mikehardy** [comment]
@stepupithub I get an email every time you comment. Once is good...

Make it look sort of like this https://github.com/mikehardy/rnfbdemo/blob/master/PodfileRN59 except I don't keep up with the versions there, it's too old to spend the time. That said, if you are on react-native-firebase 5.5.6, I think Pods 6.13.0 would work just fine

**Comment by stepupithub** [comment]
owe !!! i'm so sorry mike 
ok i dont need to specify version for pod right  it means 6.14.0 the latest one is ok with react-native-firebase 5.5.6 and react-native 0.59.10 
RNHTMLtoPDF i have changed there because it was also using UIWebView and one more thing i'm using react-native-gesture-handler 1.1.3 and "grep" command also shows me that RNGS using UIWebView but i can't find it...plz can you tell me for that which version i can use for react-native 59.10....
appreciate your hard work @mikehardy 
thank you for reply 
have a good time

**Comment by mikehardy** [comment]
As a best practice always specify versions or your build is not reproducible
I have no idea on specific libraries, but in general I would find the latest one and specify that. It should clean things up, but you may need to move forward to current versions of react-native (google "react-native upgrade-helper" to get a site that can help a lot with the upgrade). This is just for react-native-firebase issues, and we're well past UIWebView issues here so there's no relevant work for this module
